framework/Models.py

```diff
@@ -1529,44 +1534,197 @@ def run(self,inputFiles,jobHandler):
+    if returnCode == 0:
+
+      returnDict = {}
+      ## This may be a tautology at this point --DPM 4/12/17
+      if outputFile is not None:
+        outFile = Files.CSV()
+        ## Should we be adding the file extension here?
+        outFile.initialize(outputFile+'.csv',self.messageHandler,path=metaData['subDirectory'])
+
+        csvLoader = CsvLoader(self.messageHandler)
+        csvData = csvLoader.loadCsvFile(outFile)
+        headers = csvLoader.getAllFieldNames()
+
+        ## Numpy by default iterates over rows, thus we transpose the data and
+        ## zip it with the headers in order to do store it very cleanly into a
+        ## dictionary.
+        for header,data in zip(headers, csvData.T):
+          returnDict[header] = data
+
+      self._replaceVariablesNamesWithAliasSystem(returnDict, 'input', True)
+      self._replaceVariablesNamesWithAliasSystem(returnDict, 'output', True)
+
+      ## The last thing before returning should be to delete the temporary log
+      ## file and any other file the user requests to be cleared
+      if deleteSuccessfulLogFiles:
+        self.raiseAMessage(' Run "' +kwargs['prefix']+'" ended smoothly, removing log file!')
+        if os.path.exists(codeLogFile):
+          os.remove(codeLogFile)
+
+      ## Check if the user specified any file extensions for clean up
+      for fileExt in fileExtensionsToDelete:
+        if not fileExt.startswith("."):
+          fileExt = "." + fileExt
+
+        fileList = [ f for f in os.listdir(metaData['subDirectory']) if f.endswith(fileExt) ]
+
+        for f in fileList:
+          os.remove(f)
+
+      return (kwargs['SampledVars'],returnDict)
+    else:
+      self.raiseAMessage(" Process Failed "+str(command)+" returnCode "+str(returnCode))
+      absOutputFile = os.path.join(sampleDirectory,outputFile)
+      if os.path.exists(absOutputFile):
+        self.raiseAMessage(repr(open(absOutputFile,"r").read()).replace("\\n","\n"))
+      else:
+        self.raiseAMessage(" No output " + absOutputFile)
 
-  def collectOutput(self,finishedjob,output,options=None):
+      ## If you made it here, then the run must have failed
+      return None
+
+  def collectOutput(self,finishedJob,output,options=None):
     """
       Method that collects the outputs from the previous run
       @ In, finishedJob, InternalRunner object, instance of the run just finished
       @ In, output, "DataObjects" object, output where the results of the calculation needs to be stored
       @ In, options, dict, optional, dictionary of options that can be passed in when the collect of the output is performed by another model (e.g. EnsembleModel)
       @ Out, None
     """
-    outputFilelocation = finishedjob.getWorkingDir()
-    attributes={"inputFile":self.currentInputFiles,"type":"csv","name":os.path.join(outputFilelocation,finishedjob.output+'.csv')}
-    attributes['alias'] = self.alias
-    metadata = finishedjob.getMetadata()
+    if finishedJob.getEvaluation() == -1:
```

Comment:
Ooh, I like that idea. I can make a more or less empty class for now that is used here. The question is does this make sense to inherit from Exception or just object?  In this context, we are not going to raise an exception, but rather just return it and test if it is of this type in the caller.